activesupport/test/inflector_test.rb

```diff
@@ -404,6 +404,30 @@ def test_inflector_locality
     assert !ActiveSupport::Inflector.inflections.singulars.empty?
   end
 
+  def test_inflector_for_default_locale
+    ActiveSupport::Inflector.inflections(:es) do |inflect|
+      inflect.plural(/$/, 's')
+      inflect.plural(/z$/i, 'ces')
+
+      inflect.singular(/s$/, '')
+      inflect.singular(/es$/, '')
+
+      inflect.irregular('el', 'los')
+    end
+
+    I18n.available_locales << [:es]
+    I18n.locale = :es
+
+    assert_equal('hijos', 'hijo'.pluralize)
+    assert_equal('luces', 'luz'.pluralize)
+    assert_equal('sociedad', 'sociedades'.singularize)
+    assert_equal('los', 'el'.pluralize)
+  end
+
+  def teardown
+    I18n.locale = :en
```

Comment:
I think, this should be wrapped in `ensure` block in the above test case itself and we should also remove `:es` locale from available_locales list.